### ISSUE TITLE
Synchronously calling resume() leads to crash

### ISSUE THREAD

**Issue by Sebmaster** [inikulin/parse5#98]
See this small repro

```js
var parse5 = require("parse5");

const instance = new parse5.ParserStream({
  locationInfo: true
});
instance.on("script", (el, docWrite, resume) => {
  resume();
});

instance.end('<!doctype html><script>abc</script>');
```

Stack trace:
```
_stream_writable.js:346
  cb();
  ^

TypeError: cb is not a function
    at afterWrite (_stream_writable.js:346:3)
    at nextTickCallbackWithManyArgs (node.js:496:18)
    at process._tickCallback (node.js:394:17)
    at Function.Module.runMain (module.js:431:11)
    at startup (node.js:139:18)
    at node.js:999:3
```

**Comment by thisconnect** [comment]
Me to on Node.js 5.4.1 parse5@2.1.2

**Comment by inikulin** [comment]
Hm, can't reproduce on my side with node v4.2.2 and v5.4.1. Maybe, something OS specific. On which OS are you, guys?

**Comment by Sebmaster** [comment]
Ubuntu 15.10, Node.js 5.4.1 parse5@2.1.2 for me.

**Comment by inikulin** [comment]
Reproduced on both Windows and MacOS. Funny fact: this exact code doesn't throw in context of mocha test.

**Comment by inikulin** [comment]
Finishing test on `nextTick` did the trick. Seems like mocha forces process termination

**Comment by Sebmaster** [comment]
Mocha explicitly calls process.exit(1), you can call it with `--no-exit` (for example in mocha.opts) and it won't do that though.

**Comment by inikulin** [comment]
This option is not available in `gulp-mocha`, at least it's not documented. Anyway `nextTick` workaround is just fine.

**Comment by inikulin** [comment]
Published as v2.1.3

**Comment by Sebmaster** [comment]
Hrm, sadly this is breaking the assumption that you can call ParserStream synchronously, which is required in our current API. Is there no option to keep this synchronous? Might have to pull the jsdom API rework ahead...

**Comment by inikulin** [comment]
@Sebmaster It complicates things a little bit, but I have some ideas.

**Comment by Sebmaster** [comment]
:heart: :+1:

**Comment by inikulin** [comment]
@Sebmaster Fixed and published as v2.1.4

**Comment by Sebmaster** [comment]
Awesome! Works! :+1:
